src/phasorpy/phasor.py

```diff
@@ -0,0 +1,507 @@
+    (0..., 0...)
+    >>> phasor_from_lifetime(80.0, [3.9788735, 0.9947183], [0.5, 0.4])
+    (0..., 0...)
+    >>> phasor_from_lifetime(
+    ...     80.0, [3.9788735, 0.9947183], [0.5, 0.5], is_preexp=True
+    ... )
+    (0..., 0...)
+
+    """
+    tau = numpy.array(lifetime, dtype=numpy.float64, copy=True)
+    if tau.ndim > 1:
+        raise ValueError('lifetime must be scalar or one-dimensional array')
+    if fraction is None:
+        frac = numpy.ones_like(tau)
+    else:
+        frac = numpy.array(fraction, dtype=numpy.float64, copy=True)
+        if tau.shape != frac.shape:
+            raise ValueError(f'shape mismatch {tau.shape} != {frac.shape}')
+    if is_preexp:
+        # preexponential amplitudes to fractional intensities
+        frac *= tau
+    frac /= numpy.sum(frac)  # TODO: check for zero
+    tau *= frequency * 2e-3 * math.pi  # omega_tau
+    tmp = numpy.square(tau)
+    tmp += 1.0
+    tmp **= -1
+    tmp *= frac
+    re = numpy.sum(tmp)
+    tmp *= tau
+    im = numpy.sum(tmp)
+    return re, im
+
+
+def phasor_center(
+    real: ArrayLike,
+    imag: ArrayLike,
+    /,
+    *,
+    skip_axes: tuple[int, ...] | None = None,
+    method: Literal['mean', 'spatial_median', 'geometric_median'] = 'mean',
+) -> tuple[NDArray[Any], NDArray[Any]]:
+    """Return center of phasor coordinates.
+
+    Parameters
+    ----------
+    real : array_like
+        Scalar or array containing the real components of the
+        phasor coordinates.
+    imag : array_like
+        Scalar or array containing the imaginary components of the
+        phasor coordinates.
+    skip_axes : tuple[int, ...] or None, optional
+        Axes to be excluded during center calculation. If None, all
+        axes are considered.
+    method : str, optional
+        Method used for center calculation:
+            - 'mean': Arithmetic mean of the coordinates.
+            - 'spatial_median': Spatial median of the coordinates.
+            - 'geometric_median': Geometric median of the coordinates.
+
+    Returns
+    -------
+    tuple[NDArray[Any], NDArray[Any]]
+        Real and imaginary center coordinates calculated based on
+        the specified method.
+
+    Examples
+    --------
+    >>> real_data = numpy.array([1.0, 2.0, 3.0])
+    >>> imag_data = numpy.array([4.0, 5.0, 6.0])
+    >>> center_mean, center_spatial_median, center_geometric_median = (
+    ...     phasor_center(real_data, imag_data, method='mean'),
+    ...     phasor_center(real_data, imag_data, method='spatial_median'),
+    ...     phasor_center(real_data, imag_data, method='geometric_median')
+    ... )
+    >>> center_mean
+    (2.0, 5.0)
+    >>> center_spatial_median
+    (2.0, 5.0)
+    >>> center_geometric_median
+    (2.0, 5.0)
+    """
+    real = numpy.asarray(real)
+    imag = numpy.asarray(imag)
+    if skip_axes is not None:
+        real = numpy.delete(real, skip_axes, axis=None)
+        imag = numpy.delete(imag, skip_axes, axis=None)
```

Comment:
This is the part of the code that I removed because it did not have the behaviour I expected. I would need some examples of use cases to test how to implement it.